src/compiler/builder.ts

```diff
@@ -1377,10 +1378,17 @@ function getBuildInfo(state: BuilderProgramStateWithDefinedProgram): BuildInfo {
     };
     return buildInfo;
 
-    function relativeToBuildInfoEnsuringAbsolutePath(path: string) {
+    function relativePathToBuildInfo(path: string) {
         return relativeToBuildInfo(getNormalizedAbsolutePath(path, currentDirectory));
     }
 
+    function relativePathToBuildInfoOrOriginalValue(path: unknown): string | undefined {
+        if (typeof path === "string") {

```

Comment:
I'm not entirely sure what to do here. Our options parser does discard invalid values. I wanted to make the most minimal change to `builder.ts` on this one.